File: MainActivity.java
package com.example.identificador; // Verifica que coincida con tu paquete

import android.content.Intent;
import android.hardware.fingerprint.FingerprintManager;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private TextView textView;
    private ImageView imageView;
    private FingerprintManager fingerprintManager;
    private FingerprintManager.AuthenticationCallback authenticationCallback;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Enlace de vistas con layout XML
        textView = findViewById(R.id.textView2); // Asegúrate que el ID coincida con tu XML
        imageView = findViewById(R.id.imageView);

        // Inicialización del servicio biométrico
        fingerprintManager = (FingerprintManager) getSystemService(FINGERPRINT_SERVICE);

        // Configuración de los 4 métodos de respuesta biométrica
        authenticationCallback = new FingerprintManager.AuthenticationCallback() {
            @Override
            public void onAuthenticationError(int errorCode, CharSequence errString) {
                super.onAuthenticationError(errorCode, errString);
                textView.setText("ERROR: Intente de nuevo");
                imageView.setImageResource(R.drawable.failed);
            }

            @Override
            public void onAuthenticationHelp(int helpCode, CharSequence helpString) {
                super.onAuthenticationHelp(helpCode, helpString);
                textView.setText("AYUDA: Vuelve a escanear de nuevo");
                imageView.setImageResource(R.drawable.cargando);
            }

            @Override
            public void onAuthenticationSucceeded(FingerprintManager.AuthenticationResult result) {
                super.onAuthenticationSucceeded(result);
                textView.setText("¡Escaneo de huella dactilar exitoso! \nIniciando sesión…");
                imageView.setImageResource(R.drawable.correct);
                
                // Transición a la pantalla de bienvenida
                startActivity(new Intent(MainActivity.this, Resultado.class));
            }

            @Override
            public void onAuthenticationFailed() {
                super.onAuthenticationFailed();
                textView.setText("Escaneo fallido, huella dactilar no registrada");
                imageView.setImageResource(R.drawable.failed);
            }
        };
    }

    // Método asignado al botón de escaneo en el atributo onClick del XML
    public void scanButton(View view) {
        if (fingerprintManager != null) {
            try {
                fingerprintManager.authenticate(null, null, 0, authenticationCallback, null);
            } catch (SecurityException e) {
                e.printStackTrace();
            }
        }
    }
}

File: MainActivity.kt
package com.example.appaudioramses

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import android.widget.ImageView
import android.widget.TextView
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.content.Intent
import android.os.Looper
import android.os.Handler


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        // Agregar animaciones
        val animacion1 = AnimationUtils.loadAnimation(this, R.anim.desplazamiento_arriba)



        val animacion2 = AnimationUtils.loadAnimation(this, R.anim.desplazamiento_arriba)

        //val animacion1 = AnimationUtils.loadAnimation(this, R.anim.desplazamiento_abajo)
        //val animacion2 = AnimationUtils.loadAnimation(this, R.anim.desplazamiento_abajo)

        val audiotextView = findViewById<TextView>(R.id.audiotextView)
        val logoimageview = findViewById<ImageView>(R.id.logoimageview)

        audiotextView.animation = animacion1
        logoimageview.animation = animacion2
        Handler(Looper.getMainLooper()).postDelayed({

            val intent = Intent(this, ViewActivity::class.java)
            startActivity(intent)
            finish()

        },4000)


        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }
}
